pytests/conftest.py

```diff
@@ -53,39 +57,31 @@ def make_cert(self):
         cert.get_subject().emailAddress = "tdnf-devel@vmware.com"
         cert.set_serial_number(0)
         cert.gmtime_adj_notBefore(0)
-        cert.gmtime_adj_notAfter(365*24*60*60)
+        cert.gmtime_adj_notAfter(365 * 24 * 60 * 60)
         cert.set_issuer(cert.get_subject())
         cert.set_pubkey(k)
         cert.sign(k, "sha512")
         with open("cert.pem", "wt") as f:
             f.write(crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode("utf-8"))
         with open("key.pem", "wt") as f:
             f.write(crypto.dump_privatekey(crypto.FILETYPE_PEM, k).decode("utf-8"))
-        self.keyfile = os.path.join(os.getcwd(), 'key.pem')
-        self.certfile = os.path.join(os.getcwd(), 'cert.pem')
-
-    def run(self):
-        if self.enable_https:
-            self.make_cert()
-        os.chdir(self.root)
-        try:
-            self.httpd = HTTPServer(self.addr, SimpleHTTPRequestHandler)
-            if self.enable_https:
-                self.httpd.socket = ssl.wrap_socket(self.httpd.socket,
-                                                    keyfile = self.keyfile,
-                                                    certfile = self.certfile,
-                                                    server_side = True)
-            self.httpd.serve_forever()
-        finally:
-            self.httpd.server_close()
-
-    def join(self):
-        self.httpd.shutdown()
-        super().join()
+        keyfile = os.path.join(os.getcwd(), 'key.pem')
+        certfile = os.path.join(os.getcwd(), 'cert.pem')
```

Comment:
This is odd - we set `keyfile` and `certfile` in line 45/46, then open the files *without* using the variables, then assign the same values again. I know this isn't new, but should be rewritten. Also, assigning those values to local variables in the if clause (instead of attributes to `self`) in lines 45/46 doesn't seem to make sense if we are returning. Please make sure those attributres aren't used any where.